### DISCUSSION TITLE
WOCHE 05 - Demo

### DISCUSSION THREAD

**Discussion by mausio** [mausio/corgi-shop-doc#7] [Blogs]
**Diese Woche wurde das Front End weiterentwickelt, sowie das Back End.**

1. Im Backend haben sich Hürden gebildet die überwunden werden mussten, wie z.B. die Inkompatibilität von Types und die Länge von Strings; Tiefe Informationen folgen in der nächsten Woche von Mael.
2. Das Frontend beinhaltet nun eine statische Version vom Merch und die Möglichkeit auf die einzelnen Kategorien zu clicken. 
Wenn das Backend die Categories und die jeweiligen Items über drei Endpunkte zur Verfügung stellt, wird das Frontend bezüglich des DetailViews der Items weiterentwickelt. 

**Nachstehend ist der momentane Stand des Frontends visualisiert in einem GIF und verschiedenen Bildern.**

- [GIF der App](https://github.com/mausio/corgi-shop-doc/blob/main/demos/corgi-demo.gif)
Dieses dient zur Demonstration der aktuellen Nutzung der App.

- [Screenshot der Homepage](https://github.com/mausio/corgi-shop-doc/blob/main/images/demo/home.png)
Diese Seite erscheint beim Öffnen der App , später sollen hier das User-Profil, Vorschläge zu Corgis und der momentane Cart mit kurzen Informationen angezeigt werden.

- [Screenshot des Carts](https://github.com/mausio/corgi-shop-doc/blob/main/images/demo/cart.png)
Hier wird es möglich sein die Anzahl der Items zu manipulieren, diese einzeln zu löschen, sowie den Betrag zzgl. der Mehrwertsteuer zu sehen, und den Einkauf abzuschließen.

- [Screenshot des CorgisCategoryViews](https://github.com/mausio/corgi-shop-doc/blob/main/images/demo/single-category.png)
Hier soll später eine Liste von Corgis zu sehen sein, wobei man auf die Corgis einzeln clicken kann und deren ausführliche Beschreibung erhält.

- [Screenshot des CorgiSingleViews](https://github.com/mausio/corgi-shop-doc/blob/main/images/demo/single-corgi.png)
Hier soll man einzelne Corgis betrachten können und somit Informationen zu deren Preis, Gewicht und Größe erhalten.

- [Screenshot des MerchViews](https://github.com/mausio/corgi-shop-doc/blob/main/images/demo/merch-list.png)
Beim scrollen ist jetzt es möglich eine kurze Vorschau von 8 Items zu erhalten. 
Oben sollen hier die Categories zum Schnellzugriff präsentiert werden. 

**Es folgen noch:**
- im Frontend der MerchDetailView für einzelne Items
- Anfänge der HomePage. 
Weiteres folgt im zweiten Semester.

**Es haben folgende Aufgaben übernommen:**
- Front End (Robin): Swift, iOS Development, BlogPosts, Projektleitung
- Back End (Niklas): Java, SpringBoot, Corgis, Database, UML Diagramme
- Back End (Mael): Java, SpringBoot, Merch, BlogPosts, Kommentare

**Comment by Manuel-F-04** [comment]
Das Frontend sieht echt gut aus und die Idee mit GIF verdeutlicht den Aufbau deutlich. Zusätzlich finde ich das Konzept, kurz die zukünftigen Aufgaben zu erwähnen ziemlich nützlich für die Orientierung, sowie die kurzen Beschreibungen zu den Screenshot extrem hilfreich.
